osgp/platform/osgp-secret-management/src/main/java/org/opensmartgridplatform/secretmanagement/application/endpoints/SoapEndpointDataTypeConverter.java

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright 2020 Smart Society Services B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.secretmanagement.application.endpoints;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.tomcat.util.buf.HexUtils;
+import org.opensmartgridplatform.secretmanagement.application.domain.SecretType;
+import org.opensmartgridplatform.secretmanagement.application.domain.TypedSecret;
+import org.opensmartgridplatform.shared.exceptionhandling.OsgpException;
+import org.opensmartgridplatform.shared.exceptionhandling.TechnicalException;
+import org.opensmartgridplatform.shared.security.EncryptedSecret;
+import org.opensmartgridplatform.shared.security.EncryptionDelegate;
+import org.opensmartgridplatform.shared.security.EncryptionProviderType;
+import org.opensmartgridplatform.shared.security.Secret;
+import org.opensmartgridplatform.ws.schema.core.secret.management.SecretTypes;
+import org.opensmartgridplatform.ws.schema.core.secret.management.TypedSecrets;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+
+@Component
+public class SoapEndpointDataTypeConverter {
+
+    private static final String KEY_REFERENCE = "1"; //only one key in use
+    private final EncryptionDelegate encryptionDelegate;
+
+    public SoapEndpointDataTypeConverter(
+            @Qualifier("DefaultEncryptionDelegate") final EncryptionDelegate defaultEncryptionDelegate) {
+        this.encryptionDelegate = defaultEncryptionDelegate;
+    }
+
+    public List<SecretType> convertToSecretTypes(SecretTypes soapSecretTypes) throws OsgpException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'org.opensmartgridplatform.shared.exceptionhandling.OsgpException', as it cannot be thrown from method's body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)